### PR TITLE
User entity implementation

### PR DESCRIPTION
**Pull request by jaeyonglee3**
This PR brings in the User entity along with the appropriate methods for modifying each instance's favourite recipes and tagged recipes. 

Note that this PR only defines the User entity and **_does not_** fully implement the **_adding favourite recipes_** and **_adding tagged recipes_** use cases. These use cases will be fully implemented in follow up pull requests.

### CODE REVIEW

File: src/entity/CommonUser.java [hunk 1]
```diff
@@ -1,4 +1,65 @@
 package entity;
 
-public class CommonUser implements User {
-}
+import java.util.ArrayList;
+import java.util.HashMap;
+
+class CommonUser implements User {
+
+    private final String username;
```
Comment (kev0921) [inline src/entity/CommonUser.java:8]: Good choice of access modifiers.

File: src/entity/User.java [hunk 2]
```diff
@@ -1,4 +1,19 @@
 package entity;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+
 public interface User {
```
Comment (kev0921) [inline src/entity/User.java:6]: Follows interface segregation design principle by not defining irrelevant methods.

File: src/entity/CommonUserFactory.java [hunk 3]
```diff
@@ -1,4 +1,13 @@
 package entity;
 
 public class CommonUserFactory implements UserFactory {
-}
+    /**
```
Comment (kev0921) [inline src/entity/CommonUserFactory.java:4]: Write a more detailed javadoc to explain what this factory does.

File: src/entity/UserFactory.java [hunk 4]
```diff
@@ -1,4 +1,5 @@
 package entity;
 
 public interface UserFactory {
```
Comment (kev0921) [inline src/entity/UserFactory.java:3]: Follows interface segregation design principle by not defining irrelevant methods.

File: src/entity/CommonUser.java [hunk 5]
```diff
@@ -1,4 +1,65 @@
...
+    private final String password;
+    private HashMap<String, ArrayList<Recipe>> tags;
```
Comment (kev0921) [inline src/entity/CommonUser.java:10]: Could make the attribute name more specific. Maybe taggedRecipes.

File: src/entity/CommonUser.java [hunk 6]
```diff
@@ -1,4 +1,65 @@
...
+        return username;
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public HashMap<String, ArrayList<Recipe>> getAllTagged() {
```
Comment (kev0921) [inline src/entity/CommonUser.java:57]: The method name could be more specific. Maybe getTaggedRecipes. Make sure it is consistent with the instance attribute that stores the HashMap of tagged recipes.

**Review by kev0921 — CHANGES_REQUESTED** [review]
Great overall implementation of the CommonUser entity and factory. There are just some names and javadocs that need to be a little more specific, and some small tests can be created as well.

File: src/entity/CommonUser.java [hunk 7]
```diff
@@ -1,4 +1,65 @@
...
+        this.tags = new HashMap<>();
+    }
+
+    /**
+     * Assign the given recipe to the given tag. If tag does not already exist, create a new one.
+     * @param recipe the recipe to which the tag will be assigned.
+     * @param tag the tag to be given to the recipe.
+     */
+    @Override
+    public void assignTag(Recipe recipe, String tag) {
```
[the remark this answers is not in the document]
Comment (jaeyonglee3) [reply src/entity/CommonUser.java:31]: Tests will be written once all branches are merged as the Recipe entity (which is required for the recipe tags) has been implemented on another branch.

**Review by kev0921 — APPROVED** [review]
LGTM!

**Pull request merged**